Support Linux 6.9

RoyWFHuang opened the pull request:
To align with the best practices recommended by Linux kernel v6.9 API. This commit adds struct file and other related functions to support v6.9. The sturct bdev_handle is still retained for compatibility with v6.7

ref:
https://patchwork.kernel.org/project/linux-block/patch/20240427211305.GG1495312@ZenIV/#25828002 https://lore.kernel.org/all/20240123-vfs-bdev-file-v2-0-adbd023e19cc@kernel.org/

jserv commented on simplefs.h:
```diff
@@ -152,4 +129,31 @@ extern uint32_t simplefs_ext_search(struct simplefs_file_ei_block *index,
+    struct block_device *s_journal_bdev; /* v5.10+ external journal device */
+#if SIMPLEFS_AT_LEAST(6, 11, 0)
+    struct file *s_journal_bdev_file; /* v6.11 external journal device */
+#elif SIMPLEFS_AT_LEAST(6, 8, 0)
+    struct bdev_handle
+        *s_journal_bdev_handle; /* v6.7+ external journal device */
+#endif
```
Can you check if the condition fits Linux v6.7 as well?

RoyWFHuang replied:
I believe this change was introduced in v6.7.
The commit:
```
d577c8aaed2035fb7bcc970058a9d8c46c26fcaa
ext4: Convert to bdev_open_by_dev()
```
modifies the ext4 filesystem by replacing `struct block_device` with `struct bdev_handle`.
(here is the patch lists: https://lore.kernel.org/all/20230818123232.2269-1-jack@suse.cz/#r)
And this commit is included in tag v6.7
```
$ git tag --contains d577c8aae
.....
v6.14-rc5
v6.14-rc6
v6.14-rc7
v6.7
v6.7-rc1
v6.7-rc2
v6.7-rc3
v6.7-rc4
.....
```

jserv replied:
Thus, refine the use of `SIMPLEFS_AT_LEAST` macro.

RoyWFHuang committed:
Support Linux 6.9

To align with the best practices recommended by Linux kernel v6.9 API.
This commit adds struct file and other related functions to support v6.9.
The sturct bdev_handle is still retained for compatibility with v6.7

ref:
start to support struct file*
https://patchwork.kernel.org/project/linux-block/patch/20240427211305.GG1495312@ZenIV/#25828002
https://lore.kernel.org/all/20240123-vfs-bdev-file-v2-0-adbd023e19cc@kernel.org/

Move struct bdev_handle private to the block layer
https://lore.kernel.org/all/20240123-vfs-bdev-file-v2-29-adbd023e19cc@kernel.org/

jserv commented:
Thank @RoyWFHuang for contributing!
